libpromises/evalfunction.c

```diff
@@ -1586,6 +1623,44 @@ static FnCallResult FnCallGetMetaTags(EvalContext *ctx, ARG_UNUSED const Policy
         tagset = EvalContextClassTags(ctx, ref.ns, ref.name);
         ClassRefDestroy(ref);
     }
+    else if (strcmp(fp->name, "getbundlemetatags") == 0)    
+    {
+        const char *bundleref = RlistScalarValue(finalargs);
+        const Rlist *args = RlistFromSplitString(bundleref, ':');
+        const char *namespace;
+        const char *name;
+        if (!args->next)
+        {
+            namespace = "default";
+            name = RlistScalarValue(args);
+        } else {
+            namespace = RlistScalarValue(args);
+            name = RlistScalarValue(args->next);
+        }
+        const Bundle *bundle =  GetBundleFromPolicy(policy, namespace, name);
+        if (!bundle)
+        {
+            FatalError(ctx, "Couldn't find bundle '%s' with namespace '%s'", name, namespace);
```

Comment:
Returning `FnFailure()` is probably more appropriate here. `FatalError()` will stop the execution of the agent, and the remaining policy will not be evaluated. I believe `FatalError()` should only be used when it is not safe to continue execution.